src/power_grid_model/validation/rules.py

```diff
@@ -697,6 +698,33 @@ def none_missing(data: SingleDataset, component: str, fields: Union[str, List[st
     return errors
 
 
+def valid_p_q_sigma(data: SingleDataset, component: str) -> List[MultiFieldValidationError]:
+    """
+    Check if for 'sym_power_sensor' and 'asym_power_sensor' the p_sigma and q_sigma are valid.
+
+    Args:
+        data: The input/update data set for all components
+        component: The component of interest, in this case only 'sym_power_sensor' or 'asym_power_sensor'
+
+    Returns:
+        A list containing zeor or one MultiFieldValidationError, listing the p_sigma and q_sigma mismatch
+    """
+    errors = []
+    p_sigma = data[component]["p_sigma"]
+    q_sigma = data[component]["q_sigma"]
+    ids = [data[component]["id"].flatten().tolist()]
+    p_nan = np.isnan(p_sigma)
+    q_nan = np.isnan(q_sigma)
+    p_inf = np.isinf(p_sigma)
+    q_inf = np.isinf(q_sigma)
+    if np.logical_xor(p_nan.any(), q_nan.any()) or np.logical_or(p_inf.any(), q_inf.any()):
```

Comment:
i think you'll have to specify `axis=-1` to the inner `any` to ensure that it is OK to have two different components in the same scenario with different conventions (one may have `p_sigma` and `q_sigma` specified while the other may not). same for the same component across multiple scenarios

what i don't know is whether `axis=-1` will cause different behaviour between `asym_power_sensor` and `sym_power_sensor`. that would obviously be wrong

these are some nice test cases though